### DISCUSSION TITLE
Passing custom wandb params and model checkpoint callback when using train.py

### DISCUSSION THREAD

**Discussion by tleyden** [ultralytics/yolov5#10252] [Q&A]
Is the wandb experiment run accessible in order to log custom parameters?  For example, I'm running different experiments with different dataset subset sizes, how would I go about logging that information into the wandb run?

Similar question about custom checkpointing behavior - in pytorch lightning the following snippet will save the best 2 model checkpoints to wandb based on the validation loss metric, is something similar possible w/ yolov5 train.py?

```

  wandb_logger = WandbLogger(project="p")

  # log custom params
  wandb_logger.experiment.config.update(
      {
        "foo": bar, 
      }
  )

  # model checkpoint callback
  checkpoint_callback = ModelCheckpoint(
      save_top_k=2,
      monitor="val_loss_epoch",
      mode="min"
  )

  # create a pytorch lightning trainer
  trainer = pl.Trainer(
      logger=wandb_logger,   
      callbacks=[checkpoint_callback],
  )
```

**Comment by tleyden** [comment]
For logging params, I found a workaround by updating the hyperparams file:

```
# Read in the yaml
with open("yolov5/data/hyps/hyp.scratch-low.yaml", "r") as f:
  hyps = yaml.safe_load(f)
  # Add hyperparams
  hyps["param1"] = param1val

# Write the yaml back out
with open("yolov5/data/hyps/hyp.scratch-low.yaml", "w") as f:
  yaml.dump(hyps, f)
```

This will cause the values show up in the wandb UI.

I still don't see an easy way to customize the checkpointing behavior though.
